### PR TITLE
latest <- beta

### PR DESCRIPTION
**Pull request by bwp91**
feel free to not review the `/docs` folder 😆

### Changed

- Create `CHANGELOG.md` file
- Fix typos + add logo to `README.md`
- Refresh `package-lock.json` (no major changes to dep versions)
- general repo updates
- add alpha releases
- dependency updates
- fix typedoc generation
- update homebridge dependencies
- regenerate docs

### CODE REVIEW

**Comment by bwp91** [comment]
@Supereg would you be happy with a 0.12.0 release of hap-nodejs soon?

File: src/lib/Accessory.ts [hunk 1]
```diff
@@ -1183,18 +1183,21 @@ export class Accessory extends EventEmitter {
...
-   *                                new Accessory.
+   * Publishes this accessory on the local network for iOS clients to communicate with.
+   * - `info.username` - formatted as a MAC address, like `CC:22:3D:E3:CE:F6`, of this accessory.
+   *   Must be globally unique from all Accessories on your local network.
+   * - `info.pincode` - the 8-digit pin code for clients to use when pairing this Accessory.
+   *   Must be formatted as a string like `031-45-154`.
+   * - `info.category` - one of the values of the `Accessory.Category` enum, like `Accessory.Category.SWITCH`.
+   *   This is a hint to iOS clients about what "type" of Accessory this represents, so
+   *   that for instance an appropriate icon can be drawn for the user while adding a
+   *   new Accessory.
```
Comment (bauer-andreas) [inline src/lib/Accessory.ts:1194]: Thanks for improve documentation here 🚀

**Review by bauer-andreas — APPROVED** [review]
Thanks for the updates and regeneration of the docs 👍

**Comment by bauer-andreas** [comment]
> would you be happy with a 0.12.0 release of hap-nodejs soon?

@bwp91 how about we finally tag a `1.0.0` release. I caught myself endlessly waiting for the "perfect time" to tag that. But honestly I think we should just tag that now (also given that we are cutting support for older nodejs versions with this release). Having a `1.0.0` without massive changes is also great to ensure a lot of people can adopt that without encountering tons of issues.
And then we can finally embrace proper semver semantics and introduce follow-up changes step by step. Or are there any major changes you want to see before that? Or things you want to break with 1.0?

**Comment by bwp91** [comment]
I'm happy to along with any versioning that you suggest. Agreed that with a 1.0.0 we can take the opportunity to _really_ specify that the officially supported node versions are 18 and 20.

I'm not actually sure whether it would still run on e.g. node 16.

**Comment by bwp91** [comment]
>  Or are there any major changes you want to see before that? Or things you want to break with 1.0?

Nothing from me. My main focus here is to get a version of hap-nodejs published with updated dependencies, then work my way through homebridge itself and the UI with updated hap-nodejs

**Comment by bwp91** [comment]
i merged into latest branch, maybe i will await further instruction on release version!

**Comment by bauer-andreas** [comment]
Tagging a 1.0 sounds good to me 👍 

> Nothing from me. My main focus here is to get a version of hap-nodejs published with updated dependencies, then work my way through homebridge itself and the UI with updated hap-nodejs

Thanks for the hard work there 🚀

**Comment by hjdhjd** [comment]
Not with a memory leak still hanging out there…@Supereg, I’d suggest holding off for a bit there.

**Comment by bwp91** [comment]
@hjdhjd happy with a 0.12.0 release until memory leak is fixed and then bumping to 1.0.0?

**Pull request merged**